jupiter-tests/src/test/java/org/junit/jupiter/engine/ClassTemplateInvocationTests.java

```diff
@@ -1004,6 +1005,14 @@ void ignoresComposedAnnotations() {
 		assertThat(engineDescriptor.getDescendants()).isEmpty();
 	}
 
+	@Test
+	void classTemplateWithResourceLockExecutesSuccessfully() {
+		var results = executeTestsForClass(ClassTemplateWithResourceLockTestCase.class);
```

Comment:
I think we should also verify that the discovered `TestDescriptor.getExclusiveResources()` return the expected `Set<ExclusiveResource>` for the class template.

You can do that by calling `discoverTestsForClass(ClassTemplateWithResourceLockTestCase.class)` and calling `getEnginesDescriptor()` which will have a single child: the class template descriptor.

Reply:
@marcphilipp 
As suggested, I’ve added the `classTemplateWithResourceLockCollectsExclusiveResources` test.